native/textern.py

```diff
@@ -70,6 +74,26 @@ def get(self, relfn):
         with open(os.path.join(self.tmpdir, relfn), encoding='utf-8') as f:
             return f.read(), self._tmpfiles[relfn]
 
+    def add_editor(self, absfn, proc):
+        relfn = os.path.basename(absfn)
+        assert relfn in self._tmpfiles
+        self._editors[absfn] = proc
+
+    def kill_editors_configure(self, allow, timeout):
+        self.kill_editors_allow = allow
+        self.kill_editors_timeout = timeout
+
+    def kill_editors(self):
+        """Terminate editors that aren't finished yet"""
+        if not self.kill_editors_allow or not self._editors:
+            return
+        for proc in self._editors.values():
+            proc.terminate()
+        sleep(self.kill_editors_timeout)
+        # Try harder
+        for proc in self._editors.values():
+            proc.kill()
```

Comment:
A bit unsure about this. I think a cleaner approach would be to use `prctl(PR_SET_PDEATHSIG)`.

Reply:
> A bit unsure about this. I think a cleaner approach would be to use `prctl(PR_SET_PDEATHSIG)`.

I'm not familiar with `prctl`. AFAIU

1) it is not supported by stdlib; requires additional dependencies; all 3rd-party packages I found at PyPI are quite old and don't work with latest Pythons;

2) `prctl(PR_SET_PDEATHSIG)` is about threads, not processes.

Reply:
It should work to prefix the editor command with `setpriv --pdeathsig SIGTERM`.

Reply:
> It should work to prefix the editor command with `setpriv --pdeathsig SIGTERM`.

That requires a complete revamping. I'll do it in a different branch and push to a new PR.

Reply:
> It should work to prefix the editor command with `setpriv --pdeathsig SIGTERM`.

Doesn't work for me even in a simple command line test. I open a terminal, run `setpriv --pdeathsig SIGTERM gvim`, and close the terminal; `gvim` stays on screen. Shouldn't it be closed? `pkill -TERM gvim` works (`gvim` stopped) so it receives the signal.